app/src/main/java/app/revanced/integrations/shared/patches/GmsCoreSupport.java

```diff
@@ -131,16 +132,29 @@ public static void checkGmsCore(Activity mActivity) {
     /**
      * @return If GmsCore is not running in the background.
      */
+    @SuppressWarnings("deprecation")
     private static boolean contentProviderClientUnAvailable(Context context) {
         // Check if GmsCore is running in the background.
         // Do this check before the battery optimization check.
-        try (ContentProviderClient client = context.getContentResolver().acquireContentProviderClient(GMS_CORE_PROVIDER)) {
-            return client == null;
+        if (isSDKAbove(24)) {
+            try (ContentProviderClient client = context.getContentResolver().acquireContentProviderClient(GMS_CORE_PROVIDER)) {
+                return client == null;
+            }
+        } else {
+            ContentProviderClient client = null;
+            try {
+                //noinspection resource
+                client = context.getContentResolver().acquireContentProviderClient(GMS_CORE_PROVIDER);
+                return client == null;
+            } finally {
+                if (client != null) client.release();
+            }
```

Comment:
Can you test this on Android 5.0?

```suggestion
        try {
            final ContentProviderClient client = context.getContentResolver()
                    .acquireContentProviderClient(GMS_CORE_PROVIDER);
            if (client != null) {
                if (isSDKAbove(24)) {
                    client.close();
                } else {
                    client.release();
                }
                return true;
            }
        } catch (Exception ex) {
            Logger.printException(() -> "contentProviderClientUnAvailable", ex);
        }
        return false;
```